### PR TITLE
 FINERACT-428 Parallelization of Jobs :Parallelizing periodic accurual

### PR DESCRIPTION
**Pull request by wkk91193**
[Fineract-428](https://issues.apache.org/jira/browse/FINERACT-428)

Parallelizing and paging of periodic accrual

### CODE REVIEW

**Comment by vorburger** [comment]
@wkk91193 thank you very much for your interest in and contribution to Fineract/Mifos!  I'm an (old) volunteer who got reminded of Mifos/Fineract this week-end at the FOSDEM conference, and I'm taking a moment to help the project clean up some of it's old issues and pull requests etc.  

We were wondering whether you are still alive, and would be willing to `git rebase develop` this PR and take any previous review feedback into account?  If we don't hear from you within 2 weeks, we will take the liberty to close this PR without merging it, to avoid confusion for future new contributors looking at this repository - hope you understand and that's fair.

BTW I've noticed that you have raised a number of PRs last year - wondering whether you would possibly be interested to become a committer in this project in the future, and help review changes from other contributors?

/Cc FYI @apache/fineract-committers

**Comment by wkk91193** [comment]
@vorburger yes, I'm interested in becoming an Apache Committer. Besides this has no conflicts at the moment and I'd kindly request @avikganguly01 to review these changes

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -1651,29 +1651,34 @@ public LoanTermVariationsData mapRow(final ResultSet rs, @SuppressWarnings("unus
...
-                    .append(" left join m_client mc on mc.id = loan.client_id ").append(" left join m_group mg on mg.id = loan.group_id")
-                    .append(" left join m_currency curr on curr.code = loan.currency_code")
-                    .append(" left join m_loan_recalculation_details as recaldet on loan.id = recaldet.loan_id ");
-            return sqlBuilder.toString();
-        }
+        
+            public String schema() {
+                final StringBuilder sqlBuilder = new StringBuilder(400);
+                sqlBuilder
+                        .append("loan.id as loanId , mc.office_id as officeId, loan.loan_status_id as loanStatusId, ")
```
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java:1658]: it's really difficult to review this and see what actually changed here... I think you changed a lot of whitespace, and some of the SQL? Using a "trick" (append `?w=1` to the URL) I can see that - but others probably don't know that trick. Would you be willing to remove the whitespace only changes to make this easier to review? (You can submit "whitespace only" changes as separate PRs if you like.)

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java [hunk 2]
```diff
@@ -1651,29 +1651,34 @@ public LoanTermVariationsData mapRow(final ResultSet rs, @SuppressWarnings("unus
...
+                        .append("from m_loan_repayment_schedule ls INNER JOIN (SELECT l.id, l.client_id, l.product_id, l.accrued_till, l.repayment_period_frequency_enum,")
+                        .append("l.interest_calculated_from_date, l.repay_every, l.is_npa, l.currency_code, l.currency_digits, l.currency_multiplesof, l.loan_status_id ")
+                        .append("from m_loan l inner join m_client mc on mc.id = l.client_id inner join m_office o on mc.office_id = o.id  where ")
+                        .append("l.loan_status_id= ? and l.is_npa=0 and l.maturedon_date >= CURDATE() and l.accrued_till <=? and o.hierarchy like ? ")
+                        .append("and l.id >= ? limit ? ")
+                        .append(") loan on loan.id=ls.loan_id ")
+                        .append("inner join m_product_loan mpl on mpl.id = loan.product_id ")
+                        .append("inner join m_client mc on mc.id = loan.client_id ")
+                        .append("inner join m_currency curr on curr.code = loan.currency_code ")
```
Comment (avikganguly01) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java:1678]: Non-JLG group loans might not have a client ID. To be on the safer side, make it a left join m_client both for the inner and outer query.

**Review by avikganguly01 — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java (hunk 1 above)
Comment (avikganguly01) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java:1658]: Hey Michael, good to see you. Thanks for the ?w=1 tip.

**Comment by wkk91193** [comment]
@vorburger @avikganguly01 Reopened a new PR #513 based on the feedback given. Closing this one.

**Pull request closed without merging**